docs/fundamentals/write-operations.txt

```diff
@@ -258,6 +258,85 @@ An **upsert** operation lets you perform an update or insert in a single
 operation. This operation streamlines the task of updating a document or
 inserting one if it does not exist.
 
+Starting in v4.7, you can perform an upsert operation by using either of
+the following methods:
+
+- ``upsert()``: When you use this method, you can perform a **batch
+  upsert** to change or insert multiple documents in one operation.
+
+- ``update()``: When you use this method, you must specify the
+  ``upsert`` option to update all documents that match the query filter
+  or insert one document if no documents are matched. Only this upsert method
+  is supported in versions v4.6 and earlier.
+
+Upsert Method
+~~~~~~~~~~~~~
+
+The ``upsert()`` query builder method accepts the following parameters:
+
+- Array of fields and values that specify documents to update or insert.
+- Array of fields that uniquely identify documents in your first array parameter.
+- Array of fields to update if a matching document exists. If you omit
+  this parameter, {+odm-short+} updates all fields.
```

Comment:
What do you think about having the names of the arguments? It seems easier to read. This is how all PHP functions and MongoDB operators are described in the official documentation

```suggestion
The ``upsert(array $values, array|string $uniqueBy, array|null $update)`` query builder method accepts the following parameters:

- ``$values``: Array of fields and values that specify documents to update or insert.
- ``$uniqueBy``: List of fields that uniquely identify documents in your first array parameter.
- ``$update``: Optional list of fields to update if a matching document exists. If you omit
  this parameter, {+odm-short+} updates all fields.
```



"list" is more specific than array to indicate that the keys are not relevant.